add documentation (#27)
iden3/iden3.github.io: mkdocs/docs/media-types/overview.md

Media type of message defines the format of the envelope that is used in messaging. More read at [DIDComm](https://identity.foundation/didcomm-messaging/spec/#iana-media-types) specification.

###  Supported media types of envelope

-  `application/iden3-zkp-json`
-  `application/iden3comm-plain-json`
-  `application/iden3comm-signed-json`
-  `application/iden3comm-encrypted-json`

Each media type shows how envelope of the message is created and present in the payload as a `typ` field.

### When to use:
- `application/iden3-zkp-json` corresponds to jwz envelope about which you can read more [here](../proposals/jwz/overview.md). <br />
- `application/iden3comm-signed-json` corresponds to [JWS](https://www.rfc-editor.org/rfc/rfc7515.html) envelope.  <br />
- `application/iden3comm-plain-json` represents simple json message with a property 'typ'.  <br />
- `application/iden3comm-encrypted-json` is a media type for encrypted envelops  <br />

In our protocol corresponding packers are implemented in [golang](https://github.com/iden3/iden3comm/tree/main/packers) and [typescript](https://github.com/0xPolygonID/js-sdk/tree/main/src/iden3comm/packers).

> To understand which envelope to use sender of the can specify `accept` header, so responder can use one of supported profiles to authenticate.

Reviewer: ```suggestion
To determine which envelope to use, the sender can specify the `accept` header, allowing the responder to choose one of the supported profiles for authentication.
```
